tests/SemverTest.php

```diff
@@ -0,0 +1,309 @@
+    {
+        $this->assertEquals($expected, Semver::gte($version1, $version2));
+    }
+
+    /**
+     * @covers ::lt
+     * @dataProvider \Composer\Semver\Test\ComparatorTest::lessThanProvider
+     * @uses \Composer\Semver\Comparator::lessThan
+     *
+     * @param string $version1
+     * @param string $version2
+     * @param bool $expected
+     */
+    public function testLessThan($version1, $version2, $expected)
+    {
+        $this->assertEquals($expected, Semver::lt($version1, $version2));
+    }
+
+    /**
+     * @covers ::lte
+     * @dataProvider \Composer\Semver\Test\ComparatorTest::lessThanOrEqualToProvider
+     * @uses \Composer\Semver\Comparator::lessThanOrEqualTo
+     *
+     * @param string $version1
+     * @param string $version2
+     * @param bool $expected
+     */
+    public function testLessThanOrEqualTo($version1, $version2, $expected)
+    {
+        $this->assertEquals($expected, Semver::lte($version1, $version2));
+    }
+
+    /**
+     * @covers ::eq
+     * @dataProvider \Composer\Semver\Test\ComparatorTest::equalToProvider
+     * @uses \Composer\Semver\Comparator::equalTo
+     *
+     * @param string $version1
+     * @param string $version2
+     * @param bool $expected
+     */
+    public function testEqualTo($version1, $version2, $expected)
+    {
+        $this->assertEquals($expected, Semver::eq($version1, $version2));
+    }
+
+    /**
+     * @covers ::neq
+     * @dataProvider \Composer\Semver\Test\ComparatorTest::notEqualToProvider
+     * @uses \Composer\Semver\Comparator::notEqualTo
+     *
+     * @param string $version1
+     * @param string $version2
+     * @param bool $expected
+     */
+    public function testNotEqualTo($version1, $version2, $expected)
+    {
+        $this->assertEquals($expected, Semver::neq($version1, $version2));
+    }
+
+    /**
+     * @covers ::satisfies
+     * @dataProvider satisfiesProvider
+     *
+     * @param string $version
+     * @param string $constraint
+     * @param bool $expected
+     */
+    public function testSatisfies($version, $constraint, $expected)
+    {
+        $this->assertEquals($expected, Semver::satisfies($version, $constraint));
+    }
+
+    /**
+     * @return array
+     */
+    public function satisfiesProvider()
+    {
+        $positive = array_map(function ($array) {
+            $array[] = true;
+            return $array;
+        }, $this->satisfiesProviderPositive());
+
+        $negative = array_map(function ($array) {
+            $array[] = false;
+            return $array;
+        }, $this->satisfiesProviderNegative());
+
+        return array_merge($positive, $negative);
+    }
+
+    /**
+     * @return array
+     */
+    public function satisfiesProviderPositive()
+    {
+        return array(
+            array('1.2.3', '1.0.0 - 2.0.0'),
+//            array('1.2.3', '^1.2.3+build'),
+//            array('1.3.0', '^1.2.3+build'),
```

Comment:
But those aren't modifiers, it's build metadata which as per semver 
should be discarded/ignored.

AFAIK we do it for versions but apparently not for constraints.